invenio_accounts/utils.py

```diff
@@ -0,0 +1,134 @@
+
+from .errors import JWTDecodeError, JWTExpiredToken, JWTInvalidHeaderError, \
+    JWTInvalidIssuer
+
+
+def jwt_authorization_header(headers):
+    """Read the authorization token from the headers.
+
+    :param dict headers: The request headers.
+    :returns: The token.
+    :rtype: str
+    """
+    code = headers.get(
+        current_app.config['ACCOUNTS_JWT_AUTH_HEADER']
+    )
+    if code is None:
+        raise JWTInvalidHeaderError
+    return code
+
+
+def jwt_verify_token(headers):
+    """Verify the JWT token.
+
+    :param dict headers: The request headers.
+    :returns: The token data.
+    :rtype: dict
+    """
+    # Get the token from headers
+    token = jwt_authorization_header(headers)
+    # Get authentication type
+    authentication_type = current_app.config['ACCOUNTS_JWT_AUTH_HEADER_TYPE']
+    if authentication_type is not None:
+        # Get the prefix and the token
+        prefix, token = token.split()
+        # Check if the type matches
+        if prefix != current_app.config['ACCOUNTS_JWT_AUTH_HEADER_TYPE']:
+            raise JWTInvalidHeaderError
+    # Get the token data
+    decode = jwt_decode_token(code)
+    # Check the integrity of the user
+    if current_user.get_id() != decode.get('sub'):
+        raise JWTInvalidIssuer
+    return decode
+
+
+def jwt_create_token(user_id=None, additional_data=None):
+    """Encode the JWT token.
+
+    :param int user_id: Addition of user_id.
+    :param dict additional_data: Additional information for the token.
+    :returns: The encoded token.
+    :rtype: str
+
+    .. note::
+        Definition of the JWT claims:
+
+        * exp: ((Expiration Time) expiration time of the JWT.
+        * sub: (subject) the principal that is the subject of the JWT.
+        * jti: (JWT ID) UID for the JWT.
+    """
+    # Create an ID
+    uid = str(uuid.uuid4())
+    # The time in UTC now
+    now = datetime.utcnow()
+    # Build the token data
+    token_data = {
+        'exp': now + current_app.config['ACCOUNTS_JWT_EXPIRATION_DELTA'],
+        'sub': user_id or current_user.get_id(),
+        'jti': uid,
+    }
+    # Add any additional data to the token
+    if additional_data is not None:
+        token_data.update(additional_data)
+
+    # Encode the token and send it back
+    encoded_token = encode(
+        token_data,
+        current_app.config['ACCOUNTS_JWT_SECRET_KEY'],
+        current_app.config['ACCOUNTS_JWT_ALOGORITHM']
+    ).decode('utf-8')
+    return encoded_token
+
+
+def jwt_decode_token(token):
+    """Decode the JWT token.
+
+    :param str token: Additional information for the token.
+    :returns: The token data.
+    :rtype: dict
+    """
+    try:
+        return decode(
+            token,
+            current_app.config['ACCOUNTS_JWT_SECRET_KEY'],
+            algorithms=[
+                current_app.config['ACCOUNTS_JWT_ALOGORITHM']
+            ]
+        )
+    except DecodeError:
+        raise JWTDecodeError
```

Comment:
To not mask the real error I think it would be good to either:

```
# pip install future
from future.utils import raise_from

try:
    # ...
except DecodeError as exc:
    raise_from(JWTDecodeError(), exc)
# Equivalent to Python 3 version of "raise JWTDecodeError() from exc"
```